src/Search/SearchHeader.tsx

```diff
@@ -1,9 +1,48 @@
+import {StackNavigationOptions} from '@react-navigation/stack'
 import React from 'react'
+import {StyleSheet, TouchableOpacity, TouchableOpacityProps} from 'react-native'
 
+import {Icon} from '../components'
 import {SearchBar} from './SearchBar'
 import {useSearch} from './SearchContext'
 
-export interface Props {
+export const useSearchHeaderOptions = ({placeHolderText, title}) => {
+  const [searchVisible, setSearchVisible] = React.useState(false)
+  const {clearSearch} = useSearch()
+  const handleSearchClose = () => {
+    setSearchVisible(false)
+    clearSearch()
+  }
+
+  const searchHeaderOptions: StackNavigationOptions = searchVisible
+    ? {
+        headerTitleContainerStyle: styles.headerTitleContainer,
+        headerTitle: () => <SearchHeader placeholder={placeHolderText} onClose={handleSearchClose} />,
+        headerLeftContainerStyle: styles.disableFlex,
+        headerLeft: () => null,
+        headerRightContainerStyle: styles.disableFlex,
+        headerRight: () => null,
+      }
+    : {
+        headerTitleContainerStyle: styles.headerTitleContainer,
+        title: title,
+        headerRight: () => <SearchButton onPress={() => setSearchVisible(true)} />,
+        headerRightContainerStyle: {
+          ...styles.disableFlex,
+          paddingHorizontal: 16,
+          position: 'absolute',
+          right: 0,
+          bottom: 0,
+          top: 0,
+        },
+        headerLeftContainerStyle: styles.disableFlex,
+        headerLeft: () => null,
+      }
+
+  return {searchHeaderOptions}
```

Comment:
?
`return searchHeaderOptions`

Reply:
@jemubm

Reply:
i condered this here https://github.com/Emurgo/yoroi-mobile/pull/2421/files#diff-a621498cc3f31f3cebcac2ce7061d50ff62a835f415b7128bd1be251b2579b1f